internal/providers/openstack/openstack.go

```diff
@@ -91,6 +93,29 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 		return fetchConfigFromMetadataService(f)
 	})
 
+	// function literal to check both ctx.Done() as well as the amount
+	// of errors in the errChan to allow for non-blocking checks
+	watcher := func() bool {
+		select {
+		case <-ctx.Done():
+			if ctx.Err() == context.DeadlineExceeded {
+				f.Logger.Info("neither config drive nor metadata service were available in time. Continuing without a config...")
+			}
+			return true
+		default:
+			if len(errChan) > 2 {
```

Comment:
How about instead of hardcoding `2` here, we declare a `ndispatches := 0` before the first `go dispatch`, then increment it after each `go dispatch`, then here do `>= ndispatches` ? That'll convey the intent better I think.